scripts/Dockerfile

```diff
@@ -1,23 +1,28 @@
-FROM jupyter/base-notebook:python-3.9.12
-
-LABEL name="pymc"
-LABEL description="Environment for PyMC version 4"
-
+# Stage 1: Base image
+FROM jupyter/base-notebook:python-3.9.12 as base
+LABEL name="pymc" description="Environment for PyMC"
 ENV LANG=C.UTF-8 LC_ALL=C.UTF-8
-
-# Switch to jovyan to avoid container runs as root
 USER $NB_UID
 
-COPY /conda-envs/environment-dev.yml .
+# Stage 2: Install Conda dependencies
+FROM base as conda
```

Comment:
What does this line do?
Does this have something to do with the `as base` part from line 2 ?

There seeems to be a lot of back and forth with these `FROM a as b` lines..